File: sdks/dotnet/sulfone-helium/Api/Extension/Lambda.cs
using sulfone_helium.Domain.Core;

namespace sulfone_helium.Api.Extension;

public class LambdaExtension : ICyanExtension
{
    private readonly Func<IInquirer, IDeterminism, CyanExtensionInput, Task<Cyan>> _f;

    public LambdaExtension(Func<IInquirer, IDeterminism, CyanExtensionInput, Task<Cyan>> f)
    {
        _f = f;
    }
    
    public Task<Cyan> Extension(IInquirer inquirer, IDeterminism determinism, CyanExtensionInput prev)
    {
        return this._f(inquirer, determinism, prev);
    }
}

File: sdks/node/src/api/extension/lambda.ts
import type { ICyanExtension } from '../../domain/core/cyan_script.js';
import type { IInquirer } from '../../domain/core/inquirer.js';
import type { IDeterminism } from '../../domain/core/deterministic.js';
import type { CyanExtensionInput } from '../../domain/core/cyan_script_model.js';
import type { Cyan } from '../../domain/core/cyan.js';

type LambdaExtensionFn = (inquirer: IInquirer, determinism: IDeterminism, prev: CyanExtensionInput) => Promise<Cyan>;

class LambdaExtension implements ICyanExtension {
  private readonly _f: LambdaExtensionFn;

  constructor(f: LambdaExtensionFn) {
    this._f = f;
  }

  async extension(inquirer: IInquirer, determinism: IDeterminism, prev: CyanExtensionInput): Promise<Cyan> {
    return this._f(inquirer, determinism, prev);
  }
}

export type { LambdaExtensionFn };
export { LambdaExtension };
